### PR TITLE
Installers

### PR DESCRIPTION
**Pull request by danielmarbach**
@SimonCropp Some spike code which shows the v6 part of the sample for https://github.com/Particular/docs.particular.net/issues/1889 

I'm very unsure about the AD part because I've never written AD queries before.

### CODE REVIEW

**Comment by danielmarbach** [comment]
Haven't received feedback so far. Will pick this up again after my leave

**Comment by SimonCropp** [comment]
@gbiellem can u review this?

**Comment by danielmarbach** [comment]
Rebased @gbiellem @SimonCropp

**Comment by gbiellem** [comment]
@danielmarbach I'm not sure in what scenario I'd ever use SwitchInstallersByADMemberShip.  How do you see this working in practice?

File: Snippets/Core/Core_6/Installers.cs [hunk 1]
```diff
@@ -17,6 +21,65 @@ await Endpoint.Start(endpointConfiguration)
...
+            #endregion
+        }
+
+        #region InstallersRunWhenNecessaryADMembershipQuery
+        static List<string> GetGroupNames(string userName, string domain)
+        {
+            var pc = new PrincipalContext(ContextType.Domain, domain);
+            var src = UserPrincipal.FindByIdentity(pc, userName).GetGroups(pc);
+            var result = new List<string>();
+            src.ToList().ForEach(sr => result.Add(sr.UserPrincipalName));
```
Comment (gbiellem) [inline Snippets/Core/Core_6/Installers.cs]: 'UserPrincipalName" will return null for groups.  So this won't work.  The attribute you want is just 'Name'

**Pull request merged**